src/fastcs/backends/tango/dsr.py

```diff
@@ -0,0 +1,218 @@
+                    instance["fset"] = _pick_updater_fset(
+                        attr_name, attribute, single_mapping.controller
+                    )
+                    instance["access"] = AttrWriteType.WRITE
+
+            collection[d_attr_name] = server.attribute(**instance)
+
+    return collection
+
+
+def _pick_command_f(
+    method_name: str, method: Callable, controller: BaseController
+) -> Callable[..., Awaitable[None]]:
+    async def _dynamic_f(self) -> None:
+        self.info_stream(f"called {controller} f method: {method_name}")
+        return await MethodType(method, controller)()
+
+    _dynamic_f.__name__ = method_name
+    return _dynamic_f
+
+
+def _collect_dev_commands(mapping: Mapping) -> dict[str, Any]:
+    collection = {}
+    for single_mapping in mapping.get_controller_mappings():
+        path = single_mapping.controller.path
+
+        for name, method in single_mapping.command_methods.items():
+            instance = {}
+            cmd_name = name.title().replace("_", "")
+            d_cmd_name = path.upper() + "_" + cmd_name if path else cmd_name
+            instance["f"] = _pick_command_f(
+                d_cmd_name, method.fn, single_mapping.controller
+            )
+            # instance["dtype_out"] = str  # Read return string for debug
+            collection[d_cmd_name] = server.command(**instance)
+
+    return collection
+
+
+def _collect_dev_init(mapping: Mapping) -> dict[str, Callable]:
+    async def init_device(self):
+        await server.Device.init_device(self)
+        self.set_state(DevState.ON)
+        await mapping.controller.connect()
+
+    return {"init_device": init_device}
+
+
+def _collect_dev_helpers(mapping: Mapping) -> dict[str, Any]:
+    collection = {}
+
+    collection["green_mode"] = tango.GreenMode.Asyncio
+
+    return collection
+
+
+def _collect_dsr_args(options: TangoDSROptions) -> List:
+    args = []
+
+    if options.debug:
+        args.append("-v4")
+
+    return args
+
+
+class TangoDSR:
+    def __init__(self, mapping: Mapping):
+        self._mapping = mapping
+
+    def _link_process_tasks(self) -> None:
+        for single_mapping in self._mapping.get_controller_mappings():
+            _link_single_controller_put_tasks(single_mapping)
+            _link_attribute_sender_class(single_mapping)
+
+    def run(self, options: TangoDSROptions | None = None) -> None:
+        if options is None:
+            options = TangoDSROptions()
+
+        self._link_process_tasks()
+
+        dev_attributes = _collect_dev_attributes(self._mapping)
+        dev_commands = _collect_dev_commands(self._mapping)
+        dev_properties: dict = {}
+        dev_init = _collect_dev_init(self._mapping)
+        dev_helpers = _collect_dev_helpers(self._mapping)
+
+        class_body = {
+            **dev_attributes,
+            **dev_commands,
+            **dev_properties,
+            **dev_init,
+            **dev_helpers,
+        }
```

Comment:
Could be neat to use the dict [merge](https://docs.python.org/3.9/whatsnew/3.9.html#dictionary-merge-update-operators) operator here?

```suggestion
        class_body = (
            _collect_dev_attributes(self._mapping)
            | _collect_dev_commands(self._mapping)
            | _collect_dev_init(self._mapping)
            | _collect_dev_helpers(self._mapping)
        )
```
or 
```suggestion

        class_body = {}
        class_body |= _collect_dev_attributes(self._mapping)
        class_body |= _collect_dev_commands(self._mapping)
        class_body |= _collect_dev_init(self._mapping)
        class_body |= _collect_dev_helpers(self._mapping)
```